hack/build-client.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+BUILD_WORKDIR="${PWD}"
+
+function clean() {
+  rm ${BUILD_WORKDIR}/.sha256sum
+
+  rm -rf ${BUILD_WORKDIR}/pkg/client
+  mkdir ${BUILD_WORKDIR}/pkg/client
+
+  rm -rf ${BUILD_WORKDIR}/python/client
+  mkdir -p ${BUILD_WORKDIR}/python/client
+}
+
+function checksum() {
+  sha256sum swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/"$1"/redhatopenshift.json >> .sha256sum
+}
+
+function generate_golang() {
+  local API_VERSION=$1
+
+  sudo docker run \
```

Comment:
Why do we have sudo in our bash scripts? This should not need root to build.

Reply:
This is an excellent question. This part is direct copy & paste from https://github.com/Azure/ARO-RP/blob/master/Makefile#L22 I believe the `sudo` is present to make sure that docker command runs OK. To run docker without `sudo`, users have to be member of `docker` group, which not many people are by default. So I suspect this line is a workaround.

Maybe in the next run, it can be refactored to use podman instead.

Reply:
IIRC, a while ago someone told me that `sudo` is for podman (when `docker` is an alias for podman).

> Maybe in the next run, it can be refactored to use podman instead.

It is [not super convinient to run podman](https://developers.redhat.com/blog/2020/02/12/podman-for-macos-sort-of/) on macOS. If we can keep referring `docker` here that would be great.

Reply:
We are standardising on RHEL7 for prod build and hosting infrastructure because there are key MSFT components that don't support RHEL8.  This needs to stay as docker.